drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -666,9 +682,10 @@ def _mark_complete_and_publish_grade(self):
 
         # There's no going back from "completed" status to "incomplete"
         self.completed = self.completed or self._is_answer_correct() or not self.attempts_remain
-        grade = self._get_grade()
+        grade = self._calculate_grade()
         # ... and from higher grade to lower
-        if grade > self.grade:
+        current_grade = self._get_grade_if_set()
+        if current_grade is None or grade > current_grade:
```

Comment:
I LOVE the new DragNDrop.  In my testing of it on prod, I do find it confusing that once my "completion" is locked in - and then I choose to continue to experiment/test with the block, I stop getting feedback on my changes.  Any then it becomes _impossible_ for me to validate which of my changes is the correct answer.  Have you considered locking the UI once the learner "completes" the block in the "Assessment" mode - as is currently done in the "Standard" mode of the block?